cmd/soroban-cli/src/config/network.rs

```diff
@@ -178,31 +140,50 @@ impl Network {
     }
 }
 
-pub static DEFAULTS: phf::Map<&'static str, (&'static str, &'static str)> = phf_map! {
-    "local" => (
-        "http://localhost:8000/rpc",
-        passphrase::LOCAL,
-    ),
-    "futurenet" => (
-        "https://rpc-futurenet.stellar.org:443",
-        passphrase::FUTURENET,
-    ),
-    "testnet" => (
-        "https://soroban-testnet.stellar.org",
-        passphrase::TESTNET,
-    ),
-    "mainnet" => (
-        "Bring Your Own: https://developers.stellar.org/docs/data/rpc/rpc-providers",
-        passphrase::MAINNET,
-    ),
-};
+pub fn default_networks() -> &'static HashMap<String, Network> {
+    static HASHMAP: OnceLock<HashMap<String, Network>> = OnceLock::new();
 
-impl From<&(&str, &str)> for Network {
-    /// Convert the return value of `DEFAULTS.get()` into a Network
-    fn from(n: &(&str, &str)) -> Self {
-        Self {
-            rpc_url: n.0.to_string(),
-            network_passphrase: n.1.to_string(),
-        }
-    }
+    HASHMAP.get_or_init(|| {
+        let mut map = HashMap::new();
+
+        map.insert(
+            "local".to_string(),
+            Network {
+                name: "local".to_string(),
+                rpc_url: "http://localhost:8000/rpc".to_string(),
+                network_passphrase: passphrase::LOCAL.to_string(),
+            },
+        );
+
+        map.insert(
+            "futurenet".to_string(),
+            Network {
+                name: "futurenet".to_string(),
+                rpc_url: "https://rpc-futurenet.stellar.org:443".to_string(),
+                network_passphrase: passphrase::FUTURENET.to_string(),
+            },
+        );
+
+        map.insert(
+            "testnet".to_string(),
+            Network {
+                name: "testnet".to_string(),
+                rpc_url: "https://soroban-testnet.stellar.org".to_string(),
+                network_passphrase: passphrase::TESTNET.to_string(),
+            },
+        );
+
+        map.insert(
+            "mainnet".to_string(),
+            Network {
+                name: "mainnet".to_string(),
+                rpc_url:
```

Comment:
Can we print RPC url in the network resolver to get a better error message?
```
$ stellar contract info interface --id CBBJTE34KZHJACB2A6OXCRTDDLX4LI7GKYYD6XYJSGMX46BLNAG752HS --network mainnet
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.19s
     Running `target/debug/stellar contract info interface --id CBBJTE34KZHJACB2A6OXCRTDDLX4LI7GKYYD6XYJSGMX46BLNAG752HS --network mainnet`
❌ error: invalid rpc url: invalid uri character
# should be 
error: invalid rpc url (Bring Your Own: https://developers.stellar.org/docs/data/rpc/rpc-providers): invalid uri character 
```

Reply:
This error comes from soroban rpc client, so we'd need to change the upstream dependency.